code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingFullscreenMessageDelegate.java

```diff
@@ -61,8 +63,9 @@ public void onShow(final FullscreenMessage fullscreenMessage) {
     @Override
     public void onDismiss(final FullscreenMessage fullscreenMessage) {
         final InternalMessage message = (InternalMessage) fullscreenMessage.getParent();
-        if (message != null && message.getAutoTrack()) {
-            message.track(null, MessagingEdgeEventType.IN_APP_DISMISS);
+        if (message != null) {
+            message.recordEventHistory(null, MessagingEdgeEventType.DISMISS);
+            message.dismiss(false);
```

Comment:
Should dismiss be called here?
I was under the assumption that onDismiss is called after the message has been dismissed. Wouldnt this recursively call message.dismiss()